be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidator.scala

```diff
@@ -470,4 +536,88 @@ sealed class ElectionValidator(dbActorRef: => AskableActorRef) extends MessageDa
+    Await.ready(askLaoData, duration).value match {
+      case Some(Success(DbActor.DbActorReadLaoDataAck(laoData: LaoData))) =>
+        val numberOfAttendees = laoData.attendees.size
+        val correct: Boolean =
+          result.forall(electionQuestionResult =>
+            (countNumberOfVotes(electionQuestionResult) <= numberOfAttendees) && allBallotCountsArePositive(electionQuestionResult)
+          )
+        if (correct) {
+          Right(rpcMessage)
+        } else {
+          Left(error)
+        }
+      case Some(Failure(ex: DbActorNAckException)) => Left(PipelineError(ex.code, s"validation of ResultElection failed, could not retrieve laoData : ${ex.message}", rpcMessage.getId))
+      case reply                                   => Left(PipelineError(ErrorCodes.SERVER_ERROR.id, s"ElectionValidator failed : unexpected DbActor reply '$reply'", rpcMessage.getId))
+    }
+  }
+
+  private def countNumberOfVotes(electionQuestionResult: ElectionQuestionResult): Int = {
+    electionQuestionResult.result.map(_.count).sum
+  }
+
+  private def allBallotCountsArePositive(electionQuestionResult: ElectionQuestionResult): Boolean = {
+    electionQuestionResult.result.forall(electionBallotVotes => electionBallotVotes.count >= 0)
+  }
+
+  private def extractElectionQuestions(electionChannel: Channel, dbActorRef: AskableActorRef): Option[List[ElectionQuestion]] = {
+    val askForSetupElectionMessage = dbActorRef ? DbActor.ReadSetupElectionMessage(electionChannel)
+    val listOfQuestions: List[ElectionQuestion] = Await.ready(askForSetupElectionMessage, duration).value match {
+      case Some(Success(DbActor.DbActorReadAck(message))) =>
+        message.get.decodedData.get.asInstanceOf[SetupElection].questions
+      case _ => List.empty
+    }
+    if (listOfQuestions.isEmpty) {
+      None
+    } else {
+      Some(listOfQuestions)
+    }
+  }
+
+  private def extractIdQuestionAssociation(electionQuestions: List[ElectionQuestion], electionQuestionResult: List[ElectionQuestionResult]): mutable.HashMap[Hash, String] = {
+    electionQuestionResult.foldLeft(mutable.HashMap[Hash, String]())((acc, res) => acc += ((res.id, findMatchingQuestion(res.id, electionQuestions))))
+  }
+
+  private def findMatchingQuestion(id: Hash, electionQuestions: List[ElectionQuestion]): String = {
+    val matchingElectionQuestion: List[ElectionQuestion] = electionQuestions.dropWhile(question => question.id != id)
+    if (matchingElectionQuestion.isEmpty) {
+      return ""
+    }
+    matchingElectionQuestion.head.question
+
+  }
+
+  private def checkResultQuestionIds(
+      rpcMessage: JsonRpcRequest,
+      questions: List[ElectionQuestion],
+      result: List[ElectionQuestionResult],
+      electionId: Hash,
+      error: PipelineError
+  ): GraphMessage = {
+    val setupElectionIds = questions.map(_.id).toSet
+    val resultElectionIds = questions.map(_.id).toSet
+    if (setupElectionIds == resultElectionIds) {
```

Comment:
This will always be true. I guess what you wanted to do is: 
`val resultElectionIds = result.questions.map(_.id).toSet`